contracts/epochs/FlowEpoch.cdc

```diff
@@ -119,36 +131,458 @@ pub contract FlowEpochs {
+          clusterQCs: [String]) {
 
-    init () {
         self.currentEpochCounter = 0
+
+        // These values are made up
+        self.numViewsInEpoch = numViewsInEpoch
+        self.numViewsInStakingAuction = numViewsInStakingAuction
+        self.numViewsInDKGPhase = numViewsInDKGPhase
+        self.numCollectorClusters = numCollectorClusters
         self.epochMetadata = {}
+
+        let QCAdmin <- self.account.load<@FlowEpochClusterQC.Admin>(from: FlowEpochClusterQC.AdminStoragePath)
+            ?? panic("Could not load QC Admin from storage")
+
+        self.QCAdmin <- QCAdmin
+
+        let DKGAdmin <- self.account.load<@FlowDKG.Admin>(from: FlowDKG.AdminStoragePath)
+            ?? panic("Could not load DKG Admin from storage")
+
+        self.DKGAdmin <- DKGAdmin
+
+        let stakingAdmin <- self.account.load<@FlowIDTableStaking.Admin>(from: FlowIDTableStaking.StakingAdminStoragePath)
+            ?? panic("Could not load staking Admin from storage")
+
+        self.stakingAdmin <- stakingAdmin
+
+        let currentBlock = getCurrentBlock()
+
+        /// Current plan is to not emit service events for bootstrapping
+        /// but this will stay here for now until it has been completely finalized
+
+        // ///// Bootstrapping the epoch genesis /////
+        // let proposedNodes = FlowIDTableStaking.getProposedNodeIDs()
+
+        // var nodeInfoArray: [FlowIDTableStaking.NodeInfo] = []
+        // var collectorClusters: [FlowEpochClusterQC.Cluster] = []
+
+        // for id in proposedNodes {
+        //     let nodeInfo = FlowIDTableStaking.NodeInfo(nodeID: id)
+
+        //     nodeInfoArray.append(nodeInfo)
+        // }
+
+        // let stakingAuctionFinalView = currentBlock.view + self.numViewsInStakingAuction
+
+        // create first collector clusters
+        // collectorClusters = createCollectorClusters(nodeIDs: nodeInfoArray)
+
+        // emit epoch setup and epoch committed events for the genesis of epochs
+        // emit EpochSetup(counter: self.currentEpochCounter + 1 as UInt64,
+        //                 nodeInfo: nodeInfoArray,
+        //                 firstView: currentBlock.view + 1 as UInt64,
+        //                 finalView: currentBlock.view + self.numViewsInEpoch,
+        //                 collectorClusters: collectorClusters,
+        //                 randomSource: randomSource,
+        //                 DKGPhase1FinalView: stakingAuctionFinalView + self.numViewsInDKGPhase,
+        //                 DKGPhase2FinalView: stakingAuctionFinalView + (2 as UInt64 * self.numViewsInDKGPhase),
+        //                 DKGPhase3FinalView: stakingAuctionFinalView + (3 as UInt64 * self.numViewsInDKGPhase))
+
+
+        // emit EpochCommitted(counter: self.currentEpochCounter + 1 as UInt64,
+        //                     dkgPubKeys: dkgPubKeys,
+        //                     clusterQCs: clusterQCs)
+
+        let firstEpochMetadata = EpochMetadata(counter: self.currentEpochCounter,
```

Comment:
I'm not sure we need this -- is there some logic that assumes epoch metadata for the last epoch?

Reply:
no, I can get rid of it